src/JS/Movies.js

```diff
@@ -0,0 +1,30 @@
+export default class Movies {
+  constructor(type) {
+    this.baseUrl = 'https://api.themoviedb.org/3/movie/';
+    this.apiKey = '3e6e2ef61d4b19645d93d55c93096241';
+    this.type = type;
+    this.data = [];
+  }
+
+  async getData() {
+    const endpoint = `${this.baseUrl + this.type}?api_key=${this.apiKey}`;
+    await fetch(endpoint)
+      .then((response) => response.json())
+      .then((data) => {
+        this.filterData(data.results);
+      });
```

Comment:
we can use `await` instead of `.then()` to make the code easier to write and read, so check this alternative 
``` 
    const response = await fetch(endpoint);
    const data = await response.json();
    this.filterData(data.results); 
```

Reply:
We usually use await in places where we think that the result may take some time to generate.
I have used await in every call to API, ".then" is used in places that we are sure the method will return a promise and is A Perfect way to be sure that If the Method is Fullfeild or not.

Reply:
* async & await is just syntactic sugar on top of promises, 
 * await holds the execution of the code until the promise is resolved,
 * fetch() returns a promise
 *  The resolved value of the promise is treated as the return value of the await expression. 
  (.json() returns a promise.)

 so basically is the same if you use then() or await in this context, but the latter one in more cleaner.

Reply:
Thanks for your reply.